core/include/detray/masks/single3D.hpp

```diff
@@ -48,7 +53,7 @@ class single3D {
 
     /// Local coordinate frame for boundary checks
     template <typename algebra_t>
-    using local_frame_type = cartesian3<algebra_t>;
+    using local_frame_type = cartesian2<algebra_t>;
```

Comment:
I had to use `cartesian2` because `cartesian3` doen't have `projection_matrix` function. And I didn't see any problem with changing to `cartesian2`. Is it incorrect?

Also,  the coordinate systems of cartesian2 and cartesian3 are the same

Reply:
Yes so kCheckIndex can be 2 for cartesian2

Reply:
Yes, I know. My point is that I still don't like this, especially if we now start to rely on the fact that this extra information for the local-to-global is present in what is conceptually a 2D coordinate system. That feels somehow hacky. We should probably remove ```cartesian2``` and ```polar2``` in the long run

On the other hand, we don't use this shape anywhere, so it's an academic question

Reply:
Yes now the names of coordinates are weird definitely. And they are not 2D coordinate system anymore. For example, we need to change `cartesian2` to `cartesian`

Reply:
Do we really also still need ```cylinder2```? We can get ```rphi``` by a simple multiplication now

Reply:
I also wonder what single3D is for - maybe it is for TPC?